src/script/event/EventServiceNoCompound.ts

```diff
@@ -76,24 +87,24 @@ export class EventServiceNoCompound extends EventService {
         .table(StorageSchemata.OBJECT_STORE.EVENTS)
         .where('conversation')
         .equals(conversationId)
-        .and(record => {
-          const timestamp = new Date(record.time).getTime();
+        .and((record: DBEvent) => {
+          const timestamp = new Date(record.time);
           return fromCompareFunction(fromDate, timestamp) && toCompareFunction(toDate, timestamp);
         })
         .limit(limit);
     }
 
-    const records = await this.storageService.getAll(StorageSchemata.OBJECT_STORE.EVENTS);
+    const records: DBEvent[] = await this.storageService.getAll(StorageSchemata.OBJECT_STORE.EVENTS);
     return records
       .filter(record => {
-        const timestamp = new Date(record.time).getTime();
+        const timestamp = new Date(record.time);
         return (
           record.conversation === conversationId &&
           fromCompareFunction(fromDate, timestamp) &&
           toCompareFunction(toDate, timestamp)
         );
       })
-      .sort((a, b) => a.conversation - b.conversation)
+      .sort()
```

Comment:
As we are comparing strings here (conversationId) a comparator function is not necessary, as this is the default method for `sort()`

Reply:
But the `record` is of type `DBEvent` and not of type `string`?

Reply:
You are absolutely right. 🤦🏻‍♂️